### PR TITLE
Happy Thoughts API - Katharina Liebig and Erika A Porath

### PR DESCRIPTION
**Pull request by TheErry**
We did this as a pair programming project since Katharina is injured. We did some of the red and black goals. Katharina's frontend can be viewed here: https://hearts-project-happy-thoughts.netlify.app/

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 ## The problem
 
-Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
+We started writing each endpoint and testing it by using Postman. Then we added the username functionallity which required changes in both the backend and the frontend. After getting that to work we inplimented infinite scrolling by using a react package. We ran into some problems because of the poor documentation. We eventually managed to solve those by troubleshooting every step and component on it's own. 
 
 ## View it live
 
-Every project should be deployed somewhere. Be sure to include the link to the deployed project so that the viewer can click around and see what it's all about.
+Database: https://happy-nobel.herokuapp.com/
+Happy Thoughts: https://happy-nobel-59adad.netlify.app/
```
Comment (newmarkdev) [inline README.md:13]: It works great ! Well done !

File: server.js [hunk 2]
```diff
@@ -14,13 +14,89 @@ mongoose.Promise = Promise
...
+
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
 // Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+app.get('/', (_, res) => {
```
Comment (newmarkdev) [inline server.js:44]: Nice placeholder 😉

**Review by newmarkdev — APPROVED** [review]
Don't have much to add, great job on this project and  linking the api !

File: server.js [hunk 3]
```diff
@@ -14,13 +14,89 @@ mongoose.Promise = Promise
...
+    default: 0
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  },
+  userName: {
+    type: String,
+    default: "Anonymous",
+  }
```
Comment (KarolCislowski) [inline server.js:34]: nice set of fields 👍🏻

File: server.js [hunk 4]
```diff
@@ -14,13 +14,89 @@ mongoose.Promise = Promise
...
+  const perPage = 5
+  const amountOfThoughts = await Thought.estimatedDocumentCount()
+
+  const thoughts = await Thought.find()
+    .limit(perPage)
+    .skip(perPage * page)
+    .sort({
+      createdAt: "desc"
+    })
+  res.json({thoughts, amountOfThoughts})
```
Comment (KarolCislowski) [inline server.js:59]: nice effect on the frontend 👍🏻

File: server.js [hunk 5]
```diff
@@ -14,13 +14,89 @@ mongoose.Promise = Promise
...
+      },
+      {
+        $inc: {
+          hearts: 1
+        }
+      },
+      {
+        new: true
+      }
+    )
```
Comment (KarolCislowski) [inline server.js:91]: 💯

**Review by KarolCislowski — APPROVED** [review]
Great API 🦄

**Pull request closed without merging**